qiskit/primitives/statevector_estimator.py

```diff
@@ -0,0 +1,128 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Estimator class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Mapping
+from dataclasses import dataclass
+from typing import Optional, Union
+
+import numpy as np
+
+from qiskit.quantum_info import SparsePauliOp, Statevector
+
+from .base import BaseEstimatorV2
+from .containers import EstimatorPub, EstimatorPubLike, PrimitiveResult, PubResult
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class Options:
+    """Options for estimator."""
+
+    seed: Optional[Union[int, np.random.Generator]] = None
+
+    def update(self, options: Options | Mapping | None = None, **kwargs):
+        """Update the options."""
+        if options is not None:
+            if isinstance(options, Mapping):
+                options_dict = options
+            elif isinstance(options, Options):
+                options_dict = options.__dict__
+            else:
+                raise TypeError(f"Type {type(options)} is not options nor Mapping class")
+            for key, val in options_dict.items():
+                setattr(self, key, val)
+
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+
+class Estimator(BaseEstimatorV2):
+    """
+    Simple implementation of :class:`BaseEstimatorV2` with Statevector.
+    """
+
+    def __init__(self, options: Options | dict | None = None):
+        if options is None:
+            options = Options()
+        elif not isinstance(options, Options):
+            options = Options(**options)
+        self.options = options
+
+    def run(
+        self, pubs: Iterable[EstimatorPubLike], precision: float | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        job: PrimitiveJob[PrimitiveResult[PubResult]] = PrimitiveJob(self._run, pubs, precision)
+        job._submit()
+        return job
+
+    def _run(
+        self, pubs: Iterable[EstimatorPub], precision: float | None
+    ) -> PrimitiveResult[PubResult]:
+        coerced_pubs = [EstimatorPub.coerce(pub) for pub in pubs]
+
+        for pub in coerced_pubs:
+            pub.validate()
```

Comment:
FYI see #11588. When that is merged, we will be able to

```suggestion
        precision = precision or self.options.precision # TODO: switch away from options to class variable
        coerced_pubs = [EstimatorPub.coerce(pub, precision) for pub in pubs]
```